pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -388,6 +396,84 @@ func (m *topologyPairsMaps) clone() *topologyPairsMaps {
 	return copy
 }
 
+func (podSpreadMap *topologyPairsPodSpreadMap) addPod(addedPod, metapod *v1.Pod, node *v1.Node) error {
+	constraints := getHardTopologySpreadConstraints(metapod)
+	match, err := podMatchesAllSpreadConstraints(addedPod, metapod.Namespace, constraints)
+	if err != nil {
+		return err
+	}
+	if !match || !nodeLabelsMatchSpreadConstraints(node.Labels, constraints) {
+		return nil
+	}
+
+	// records which topology key(s) needs to be updated
+	minMatchNeedingUpdate := make(map[string]struct{})
+	for _, constraint := range constraints {
+		pair := topologyPair{
+			key:   constraint.TopologyKey,
+			value: node.Labels[constraint.TopologyKey],
+		}
+		// it means current node is one of the critical paths of minMatchMap[TopologyKey]
+		if int32(len(podSpreadMap.topologyPairToPods[pair])) == podSpreadMap.minMatchMap[pair.key] {
+			minMatchNeedingUpdate[pair.key] = struct{}{}
+		}
+		podSpreadMap.addTopologyPair(pair, addedPod)
+	}
+	// no need to addHeadlessTopologyPair b/c if a headless pair must be present,
+	// it should have already been created earlier in removePod() phase
```

Comment:
Do we need to calculate `podSpreadMap.podToTopologyPairs` for `addedPod`? Alternatively, do we need to remove it during removePod?

Reply:
We should, to avoid potential inconsistency. And it's not expensive.

Reply:
Please add it to this PR. Unless I'm missing it?

Reply:
@Huang-Wei Do we need this for consistency?

Reply:
@alculquicondor `func (m *topologyPairsMaps) addTopologyPair` does it in pair. Is this your question?

Reply:
Yes, sorry I missed it.